pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -532,15 +536,37 @@ def _save_top_k_checkpoints(self, trainer, pl_module, metrics):
             current = trainer.training_type_plugin.reduce(current, reduce_op="mean")
 
         if self.check_monitor_top_k(current):
-            self._update_best_and_save(current, epoch, step, trainer, pl_module, metrics)
-        elif self.monitor is not None and self.verbose:
+            self._update_best_and_save(current, epoch, step, trainer, monitor_candidates)
+        elif self.verbose:
             rank_zero_info(f"Epoch {epoch:d}, step {step:d}: {self.monitor} was not in top {self.save_top_k}")
 
+    def _save_none_monitor_checkpoint(self, trainer, monitor_candidates: Dict[str, Any]):
+        if self.monitor is not None:
```

Comment:
Shouldn't it be `self.monitor is not None and self.save_top_k == -1`

Reply:
`save_top_k == -1` means save all checkpoints. so we don't want to skip in that case

But you made me find a bug, ty! :D